pkg/codec/encodings/binary/gen/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"go/format"
+	"os"
+	"text/template"
+)
+
+func main() {
+	for _, gen := range []genInfo{
+		{template: intsTemplate, fileName: "int_gen.go"},
+		{template: intsTestTemplate, fileName: "int_gen_test.go"},
+	} {
+		t, err := template.New(gen.fileName).
+			Funcs(map[string]any{"div": func(a, b int) int { return a / b }}).
+			Parse(gen.template)
+		if err != nil {
+			panic(err)
+		}
+
+		br := bytes.Buffer{}
+		if err = t.Execute(&br, []int{8, 16, 32, 64}); err != nil {
+			panic(err)
+		}
+
+		res, err := format.Source(br.Bytes())
+		if err != nil {
+			panic(err)
+		}
+
+		res = []byte(
+			fmt.Sprintf(
+				"// DO NOT MODIFY: automatically generated from pkg/codec/raw/types/main.go using the template %s\n\n%s",
```

Comment:
Is this file name in sync? I think you could extract it from the runtime caller info to keep it up to date.